compile error: cannot type error as type *roachpb.Error in return argument

Running `make test PKG=./sql`, I get:

```
go test -v  -i ./sql
go test  -run . -timeout 1m10s ./sql 
# github.com/cockroachdb/cockroach/sql_test
sql/main_test.go:158: cannot use err (type error) as type *roachpb.Error in return argument
FAIL	github.com/cockroachdb/cockroach/sql [build failed]
make: *** [test] Error 2
```

The cause is that in function `checkEndTransactionTrigger`, an error is returned for `*roachpb.Error`.

--- a/sql/main_test.go
+++ b/sql/main_test.go
@@ -155,7 +155,7 @@ func checkEndTransactionTrigger(args storageutils.FilterArgs) *roachpb.Error {
 	for _, span := range req.IntentSpans {
 		keyAddr, err := keys.Addr(span.Key)
 		if err != nil {
-			return err
+			return roachpb.NewError(err)
 		}
 		if bytes.Compare(keyAddr, keys.SystemConfigSpan.Key) >= 0 &&
 			bytes.Compare(keyAddr, keys.SystemConfigSpan.EndKey) < 0 {

